x/stakeibc/keeper/msg_server_fast_unbond.go

```diff
@@ -0,0 +1,117 @@
+package keeper
+
+import (
+	"context"
+	"fmt"
+	epochtypes "github.com/Stride-Labs/stride/v5/x/epochs/types"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+
+	"github.com/Stride-Labs/stride/v5/x/stakeibc/types"
+)
+
+func (k msgServer) FastUnbond(goCtx context.Context, msg *types.MsgFastUnbond) (*types.MsgFastUnbondResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	k.Logger(ctx).Info(fmt.Sprintf("fast unbond: %s", msg.String()))
+
+	// get our addresses, make sure they're valid
+	sender, err := sdk.AccAddressFromBech32(msg.Creator)
+	if err != nil {
+		return nil, sdkerrors.Wrapf(sdkerrors.ErrInvalidAddress, "creator address is invalid: %s. err: %s", msg.Creator, err.Error())
+	}
+
+	// then make sure host zone is valid
+	hostZone, found := k.GetHostZone(ctx, msg.HostZone)
+	if !found {
+		return nil, sdkerrors.Wrapf(types.ErrInvalidHostZone, "host zone is invalid: %s", msg.HostZone)
+	}
+
+	// get the coins to return, they need to be in the format {amount}{denom}
+	// is safe. The converse is not true.
+	ibcDenom := hostZone.GetIbcDenom()
+	coinString := msg.Amount.String() + ibcDenom
+	outCoin, err := sdk.ParseCoinNormalized(coinString)
+	if err != nil {
+		k.Logger(ctx).Error(fmt.Sprintf("failed to parse coin (%s)", coinString))
+		return nil, sdkerrors.Wrapf(err, "failed to parse coin (%s)", coinString)
+	}
+	// construct desired unstaking amount from host zone
+	stDenom := types.StAssetDenomFromHostZoneDenom(hostZone.HostDenom)
+	nativeAmount := sdk.NewDecFromInt(msg.Amount).Mul(hostZone.RedemptionRate).RoundInt()
+
+	if nativeAmount.GT(hostZone.StakedBal) {
+		return nil, sdkerrors.Wrapf(types.ErrInvalidAmount, "cannot unstake an amount g.t. staked balance on host zone: %v", msg.Amount)
+	}
+
+	// safety check: redemption rate must be within safety bounds
+	rateIsSafe, err := k.IsRedemptionRateWithinSafetyBounds(ctx, hostZone)
```

Comment:
Both LiquidStake and RedeemStake have this same check in similar places.  From a purely functional perspective, it seems mostly immaterial to me.  What would be the rationale for doing this?

Reply:
Agreed, functionally immaterial. I was thinking it would be nice break up the logical flow of the function to
1) safety checks
2) setting up the coin variables
3) executing the coin-related logic

Open to alternatives though! What do you think?

Reply:
That makes sense.  I have edited the code to reflect that.  Let me know what you think.